src/gui/editors/BBEditor.cpp

```diff
@@ -326,11 +326,15 @@ void BBTrackContainerView::clonePattern()
 
 	BBTrack *bbt = BBTrack::findBBTrack(cur_bb);
 
-	// Clone the track
-	Track *newTrack = bbt->clone();
-
-	// Track still have the TCOs which is undesirable in this case, clear the track
-	newTrack->lock();
-	newTrack->deleteTCOs();
-	newTrack->unlock();
+	if( bbt )
+	{
+		// Clone the track
+		Track *newTrack = bbt->clone();
+		bbtc->setCurrentBB( dynamic_cast<BBTrack *>( newTrack )->index() );
```

Comment:
```suggestion
		bbtc->setCurrentBB( static_cast<BBTrack *>( newTrack )->index() );
```
I think you can use a static cast in this case. We know that `bbt` is a BBTrack, and since `newTrack` is cloned from `bbt` it should have the same type.